### PR TITLE
MPG-task

### PR DESCRIPTION
**Pull request by the-darkesttt**
### Memory Pair Game task

[Demo](https://the-darkesttt.github.io/memory-pair-game/) |
[Code base](https://github.com/the-darkesttt/memory-pair-game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@the-darkesttt looks nice. 
This is not a code review though.

How about posting a link to the demo in the students' chat and collecting feedback re usability etc?

**Comment by OleksiyRudenko** [comment]
@the-darkesttt very smooth gameplay and a lot better UX. I like it.

Disclaimer: This is not a code review though.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!

File: submissions/the-darkesttt/memory-pair-game/js/newApp.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+function shuffleCards(cards) {
+    cards.sort(() => 0.5 - Math.random());
+}
+
+function createCards() {
+  cardSprites.forEach((sprite) => {
+    cardsItems += createCard(sprite);
+  });
+}
```
Comment (A-Ostrovnyy) [inline submissions/the-darkesttt/memory-pair-game/js/newApp.js:125]: Get a full string of cards using the `reduce` array method and pass it directly to the destination(line 132). With this approach variable `cardsItems` is redundant.

File: submissions/the-darkesttt/memory-pair-game/js/newApp.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  foundedPairs = 0;
+  sec = 0;
+  min = 0;
+  flippedCardsArr = [];
+  stopGame = false;
+
+  createCardDeck();
+}
+
+let cardSprites = [
```
Comment (A-Ostrovnyy) [inline submissions/the-darkesttt/memory-pair-game/js/newApp.js:76]: Move all constants and variable declarations to the top of the file. if they are on the root level.

File: submissions/the-darkesttt/memory-pair-game/js/newApp.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    if (sec === 59) {
+      sec = 0;
+      min++;
+    }
+
+    setTimeout(countGameTime, 1000);
+  }
+}
+
+const restartButton = document.querySelector('#restart-btn');
```
Comment (A-Ostrovnyy) [inline submissions/the-darkesttt/memory-pair-game/js/newApp.js:63]: Move all constants and variable declarations to the top of the file. if they are on the root level.

File: submissions/the-darkesttt/memory-pair-game/js/newApp.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        id: 'spite-05',
+        src: 'img/05.webp',
+    },
+    {
+        id: 'spite-06',
+        src: 'img/06.webp',
+    },
+];
+
+function createCard(card) {
```
Comment (A-Ostrovnyy) [inline submissions/the-darkesttt/memory-pair-game/js/newApp.js:103]: Combine it with `createCards` function

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good progress!

**Comment by the-darkesttt** [comment]
Thank you for your suggestions!

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Pull request merged**